graphql/chat/mutations.ts

```diff
@@ -1,37 +1,70 @@
-import { Arg, Ctx, Mutation, Resolver } from 'type-graphql';
-import { Lecture as Appointment } from '../generated';
+import { Arg, Authorized, Ctx, Mutation, Resolver } from 'type-graphql';
 import { Role } from '../../common/user/roles';
+import * as GraphQLModel from '../generated/models';
 import { GraphQLContext } from '../context';
 import { AuthorizedDeferred, hasAccess } from '../authorizations';
 import { getLogger } from '../../common/logger/logger';
 import { prisma } from '../../common/prisma';
+import { ConversationInfos, getOrCreateChatUser, getOrCreateConversation } from '../../common/chat';
+import { getUser } from '../../common/user';
+import { getMatchByMatchees } from '../../common/chat/helper';
 
 const logger = getLogger('MutateChatResolver');
-
-@Resolver(() => Appointment)
+@Resolver()
 export class MutateChatResolver {
     @Mutation(() => Boolean)
     @AuthorizedDeferred(Role.OWNER)
-    async matchChatCreate(@Ctx() context: GraphQLContext, @Arg('matchId') matchId: number) {
-        const match = await prisma.match.findUnique({ where: { id: matchId }, include: { pupil: true } });
+    async matchChatCreate(@Ctx() context: GraphQLContext, @Arg('matcheeUserId') matcheeUserId: string) {
+        const { user } = context;
+        const matcheeUser = await getUser(matcheeUserId);
+        const matchees = [user, matcheeUser];
+
+        const match = await getMatchByMatchees([user.userID, matcheeUserId]);
```

Comment:
(Wouldn't it be easier to pass the matchId in here instead? Or do we have a place in the UI where we have the userId but not the match?)

Reply:
we give the UI the userID and not the matchId